Restore tests merge add model tests

vanlankveldthijs opened the pull request:
Add the new tests introduced in PR #84 on top of PR #96

This should complete the suite of tests to run on the development branch.

SarahAlidoost committed:
add tests for network module

SarahAlidoost committed:
Merge branch 'master' into add_tests_network

SarahAlidoost committed:
Merge pull request #64 from SDCCA/add_tests_network

Add tests for network module

vanlankveldthijs committed:
Merge pull request #84 from SDCCA/add_tests_model

add test for model module

Commit:
Fixed variable name in _load_model.

Commit:
Temporarily disabled several failing tests. These MUST be restored before merging.

Commit:
Merge branch 'master' into restore_tests_merge_add_model_tests

Commit:
Merge branch 'development' into restore_tests_merge_add_model_tests

Commit:
Fixed two bugs in test_network.py

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -101,33 +102,36 @@ def test_set_model_parameters_with_file(self, config_file):
         assert model.model_graph is not None
         assert model.number_of_edges is not None
-        assert model.model_data['modelTheta'] is not None
+        #assert model.model_data['modelTheta'] is not None #TODO(tvl) restore assertion? model.model_data no longer exists.
```
```suggestion
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -101,33 +102,36 @@ def test_set_model_parameters_with_file(self, config_file):
         assert Path('my_model/edge_data/0.zarr').exists()
         assert str(model.model_graph.device) == 'cpu'
-        assert str(model.model_data['modelTheta'].device) == 'cpu'
+        #assert str(model.model_data['modelTheta'].device) == 'cpu' #TODO(tvl) restore assertion? model.model_data no longer exists.
```
```suggestion
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -143,7 +147,7 @@ def test_initialize_model_properties(self):
 
         modelTheta = torch.tensor([1., 1., 1., 1., 1.])
-        assert (model.model_data['modelTheta'] == modelTheta).all()
+        #assert (model.model_data['modelTheta'] == modelTheta).all() #TODO(tvl) restore assertion? model.model_data no longer exists.
```
```suggestion
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -166,14 +170,14 @@ def test_step(self, model):
 
         assert model.step_count == 1
-        assert model.model_data is not None
+        #assert model.model_data is not None #TODO(tvl) restore assertion? model.model_data no longer exists.
```
```suggestion
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -166,14 +170,14 @@ def test_step(self, model):
 
         assert model.step_count == 5
-        assert model.model_data is not None
+        #assert model.model_data is not None #TODO(tvl) restore assertion? model.model_data no longer exists.
```
```suggestion
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -195,13 +199,13 @@ def test_model_init_savestate_not_default(self, model):
     def test_model_init_restart(self, model):
         model.step_target = 3 # only run the model till step 3
```
```suggestion
        model.savestate = 1
        model.step_target = 3 # only run the model till step 3
```

SarahAlidoost replied:
In the `main` branch, the default value of `savestate` is 1. But it seems that in `development` branch is 10.

SarahAlidoost replied:
The reason for failing test is that the function `test_model_init_savestate(self, model):` still includes `assert Path('my_model/model_data.bin').exists()`. Can you please remove that line?

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -195,13 +199,13 @@ def test_model_init_savestate_not_default(self, model):
         model.step_target = 3 # only run the model till step 3
         model.run()
-        expected_generator_state = set(model.inputs["generator_state"].tolist())
+        #expected_generator_state = set(model.inputs["generator_state"].tolist())
```
```suggestion
        expected_generator_state = set(model.inputs["generator_state"].tolist())
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -195,13 +199,13 @@ def test_model_init_savestate_not_default(self, model):
         model.step_target = 5 # contiune the model till step 5
         model.run()
-        stored_generator_state = set(model.inputs["generator_state"].tolist())
+        #stored_generator_state = set(model.inputs["generator_state"].tolist())
```
```suggestion
        stored_generator_state = set(model.inputs["generator_state"].tolist())
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -195,13 +199,13 @@ def test_model_init_savestate_not_default(self, model):
         assert model.inputs is not None
-        assert model.inputs["step_count"] == 5
-        assert stored_generator_state == expected_generator_state
+        assert model.inputs["step_count"] == 4 #TODO(tvl) test: was 5 before...
+        #assert stored_generator_state == expected_generator_state #TODO(tvl) restore assertion? model.inputs no longer exists.
```
```suggestion
        assert model.inputs["step_count"] == 5
        assert stored_generator_state == expected_generator_state 
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -41,16 +41,17 @@ def test_ptm_step_timestep1(self, model):
-        global_θ =model.model_data['modelTheta'][1]
-        𝛿=model.steering_parameters['depreciation']
-        new_wealth = (global_θ + m * (1-global_θ)) * (model.model_graph.ndata['income'] - c - i_a + (1-𝛿) * k)
+    # TODO(tvl) restore test? model.model_data no longer exists.
+    # def test_ptm_step_timestep2(self, model):
+    #     model.step() # timestep 1
+    #     k,c,i_a,m = model.model_graph.ndata['wealth'],model.model_graph.ndata['wealth_consumption'],model.model_graph.ndata['i_a'],model.model_graph.ndata['m']
+    #     global_θ =model.model_data['modelTheta'][1]
+    #     𝛿=model.steering_parameters['depreciation']
+    #     new_wealth = (global_θ + m * (1-global_θ)) * (model.model_graph.ndata['income'] - c - i_a + (1-𝛿) * k)
```
```suggestion
     def test_ptm_step_timestep2(self, model):
         model.step() # timestep 1
```

SarahAlidoost replied:
after committing the suggestions, can you please check the indentation.

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -41,16 +41,17 @@ def test_ptm_step_timestep1(self, model):
-        model.step() # timestep 2
-        assert (model.model_graph.ndata['wealth'] == new_wealth).all()
-        assert Path('./my_model/edge_data/2.zarr').exists()
+    #     model.step() # timestep 2
+    #     assert (model.model_graph.ndata['wealth'] == new_wealth).all()
+    #     assert Path('./my_model/edge_data/2.zarr').exists()
```
```suggestion
         model.step() # timestep 2
         assert Path('my_model/edge_data/2.zarr').exists()
```

SarahAlidoost commented on dgl_ptm/tests/test_model.py:
```diff
@@ -41,16 +41,17 @@ def test_ptm_step_timestep1(self, model):
         assert Path('./my_model/agent_data.zarr').exists()
         assert Path('./my_model/edge_data/1.zarr').exists()
```
```suggestion
        assert Path('my_model/agent_data.zarr').exists()
        assert Path('my_model/edge_data/1.zarr').exists()
```

SarahAlidoost reviewed (changes requested):
@vanlankveldthijs thanks for the fixes :+1: . With my suggestions on the changed files, the checks should be green again.

vanlankveldthijs committed:
Apply suggestions from code review

Co-authored-by: SarahAlidoost <55081872+SarahAlidoost@users.noreply.github.com>

Commit:
Minor indentation bug fix and no longer checking for model_data path.

Commit:
Fixed bugs in test_model.py related to 0-based timesteps.

When testing in a clean environment (e.g. no my_model folder),
some assertions checking whether the edge date zarr files exist fail.
This is because the files are stored using 0-based timesteps,
while the tests assumed 1-based timesteps.
